### ISSUE TITLE
Could the `Color` animation do interpolation on custom shader property?

### ISSUE THREAD

**Issue by Jayatubi** [Demigiant/dotween#312]
It looks like the DoTween has hard coded a property named '_Color', which is usual, for the Color type animation. It would be better if this property name could be customized since I may have some custom shader with more than one color properties and I want to use the DoTween to do the interpolation on them.

UPDATE.

I found there is a `DOColor` api interface accept a `property` argument but there is no interface to set in in the inspector.

**Comment by Demigiant** [comment]
Ahhh you mean to set it from a DOTweenAnimation?

**Comment by Jayatubi** [comment]
> Ahhh you mean to set it from a DOTweenAnimation?

Exactly. So I could add some simple animation via inspector without code.

**Comment by nzeman** [comment]
Is this possible maybe now? I haven't found a way to do this.

**Comment by Demigiant** [comment]
I'm working on a new visual system which will allow this, but right now the DOTweenAnimation only works on the default "_Color" property. Is this very urgent? If it is, I could find a way to add this next week.

**Comment by nzeman** [comment]
Actually I am not in rush for the visual system, I am just wondering if this is possible to do with code. I need to change color of the material which uses _Color1 instead of _Color. Is there a possibility of having DOColor(Color from, Color to, float duration)? If I could just change from one color to another on a color (sounds funny I know) that would be awesome! Thanks for the quick reply, I love you asset and have bought Pro long time ago!

**Comment by Demigiant** [comment]
Ah with code it's doable, yes. [DOColor](http://dotween.demigiant.com/documentation.php?api=DOColor__4) has an overload which accepts a property name and another which accepts an ID :) And instead of using a dynamic from you can chain a direct From to go from color A to B:

`myMaterial.DOColor(toColor, "_Color1", duration).From(fromColor)`

**Comment by nzeman** [comment]
Awesome! Any possibility of having only color from - to?

**Comment by Demigiant** [comment]
I'm not sure I understand the question :B

**Comment by nzeman** [comment]
> I'm not sure I understand the question :B


public Color _color;
_color.DOColor(Color.black, 1f);

I've done that with DOTween.To() and doing it with r g b a properties. Anyway I've managed to make it work with the overload you've mentioned, thank you very much for your time. 👍

**Comment by GalacticMuffin** [comment]
> Ah with code it's doable, yes. [DOColor](http://dotween.demigiant.com/documentation.php?api=DOColor__4) has an overload which accepts a property name and another which accepts an ID :) And instead of using a dynamic from you can chain a direct From to go from color A to B:
> 
> `myMaterial.DOColor(toColor, "_Color1", duration).From(fromColor)`

I am also looking for a way to animate a property on a shader. but unlike the OP's question I'm not trying to access a standard shader property like a color. I have written a custom dissolve shader with a Range property that controls the disolve effect. I am looking to animate that property either directly or to set to a float property every frame within the shader which I would animate instead. i am trying to do this via a DOTween.To() but I'm a bit confused how the getter and setter would be formatted as shader properties are accessed in script with this method: Shader.SetGlobalFloat("_myValue", myValue); so there doesn't appear to be any way to get a reference to this specific shader property which i could use in a DOTween.To().

I could just not be going about this the correct way and maybe your original overloaded method you pointed out will work for this case too.

**Comment by Demigiant** [comment]
Ahoy!

Shaders are only really exposed via Materials that use them at runtime, so in your case you should use [DOFloat](http://dotween.demigiant.com/documentation.php?api=DOFloat) :)
`myMaterial.DOFloat(toValue, propertyName, duration);`

**Comment by GalacticMuffin** [comment]
Thanks for the answer! yes that's what i was thinking too and i tried using DOfloat initially but i was unable to get the property to tween. this is hard to debug do to the fact that it is passing a string for the property reference but I'm pretty sure i have the reference correct. The property i want to animate in the shader is the _Level property:

`Properties
	{
		_MainTex("Texture", 2D) = "white" {}
		_MainColor("Main Color", Color) = (1,1,1,1)
		_NoiseTex("Texture", 2D) = "white" {}
		_Level("Dissolution level", Range(0.0, 1.0)) = 0.1
	}`

and I also have it being initialized in the subshader: 
`uniform float _Level;`

perhaps there is an extra step I'm missing here to make it compatible with a DOFloat? 

here is the code in script:

`void DeathAnim()
    {
        _mat = _meshRenderer.material;
        _meshRenderer.material = dissolveMaterial;

        _mat.DOFloat(1f, "_Level", 2);
}`

**Comment by Demigiant** [comment]
I admit that Shaders programming is my Achilles' heel (though thankfully I don't need to know that for DOTween) :P To make a quick test, see if you can change the shader value via Unity's regular [SetFloat](https://docs.unity3d.com/ScriptReference/Material.SetFloat.html): if it can be changed like that then it can be animated. And if it still doesn't animate then maybe the value is at 1 already?
